src/account/operations/syncing/outputs.rs

```diff
@@ -122,40 +122,44 @@ impl AccountHandle {
         let transactions = self.read().await.transactions.clone();
         let incoming_transactions = self.read().await.incoming_transactions.clone();
 
-        let mut tasks = Vec::new();
-
-        for transaction_id in transaction_ids {
-            // Don't request known transactions again
-            if transactions.contains_key(&transaction_id) || incoming_transactions.contains_key(&transaction_id) {
-                continue;
-            }
+        // Limit parallel requests to 100, to avoid timeouts
+        for transaction_ids_chunk in transaction_ids.chunks(100).map(|x: &[TransactionId]| x.to_vec()) {
```

Comment:
you mention 10k in the issue, but pick 100 here. why did you pick this? maybe 1k still works as well?

Reply:
10k was the total amount which lead to errors, because it was doing all at once doing it in chunks of 100 worked then

Reply:
Perfect, just making sure the number wasnt arbitrary